### DISCUSSION TITLE
Artillery Pro with ECS clusters alternative to running on AWS Lambda ?

### DISCUSSION THREAD

**Discussion by anb01** [artilleryio/artillery#1919] [Artillery]
I've been using Artillery core and when running tests on local machines, the results seem variable and inconsistent based on the machine running the tests.  Assuming the network, CPU usage, etc all play a part in the results, this makes sense.  
I tried moving the tests over to run on an AWS Lambda instance with Artillery to hopefully get consistent results, regardless of who was running them.  Some of our requests are intensive so the 15-minute timeout limit for an AWS Lambda instance kept interfering and cutting off our tests.  

Would Artillery Pro's ECS ability be the answer here?  Would I be able to get consistent results, regardless of the machine running the test command, and not be limited to a 15 minute or less test?

**Comment by hassy** [comment]
hi @anb01 👋 indeed that would be the way to go. We're open sourcing ECS support soon
